godot-core/src/obj/oneditor.rs

```diff
@@ -0,0 +1,241 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::meta::{FromGodot, GodotConvert};
+use crate::registry::property::Var;
+
+// Possible areas for improvement that can be explored:
+// - Should we provide something similar to [`from_base_fn()`](crate::OnReady::from_base_fn)? In general more elaborate late initialization logic should be handled either by Option or OnReady.
+// - Adding `OnEditor` section to `init(…)`. Might be noisy and unnecessary, since OnEditor, for now, avoids elaborate late initialization logic.
+// - Should we keep "invalid" value for primitives?
+
+/// Represents an exported property that must not be null and must be set via the editor or associated code before use.
+/// Allows to use `Gd<T>`, which by itself never holds null objects, as an `#[export]` that should not be null during runtime.
+///
+/// Panics during access if value hasn't been set.
+/// Checks if value has been set before the `ready` is being run and panics if any of `OnEditor` fields is not properly initialized (this logic is exclusive to Nodes).
+/// `OnEditor<T>` should always be used as a property, preferably in tandem with an `#[export]` or `#[var]`.
+/// It should be used as it would be a value itself and lack thereof treated as a logical error.
+///
+/// A new instance can be created and have its required properties set after initialization,
+/// though [`Option<Gd<T>>`](std::option) and [`OnReady<Gd<T>>`](crate::obj::onready::OnReady) should be preferred for late initialization.
+///
+/// # Using `OnEditor<T>` with `Gd<T>` and `DynGd<T, D>`
+///
+/// Exposing properties to the Godot editor is primary use of the `OnEditor<Gd<T>>`.
+/// Default values – used in case if no value will be set via the editor – can be provided with `#[init(val = ...)]`.
+///
+/// ## Example - auto-generated init
+///
+/// ```
+///  use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct MyClass {
+///     #[export]
+///     editor_field: OnEditor<Gd<Resource>>,
+///     #[export]
+///     #[init(val = OnEditor::init(Node::new_alloc()))]
+///     required_with_default: OnEditor<Gd<Node>>,
+///     // Does **NOT** require base field to work.
+///     base: Base<Node>,
+/// }
+///
+/// #[godot_api]
+/// impl INode for MyClass {
+///     fn ready(&mut self) {
+///         // Field `required_with_default` can be either default value - specified in `#[init]` or value set via the Godot Editor.
+///        assert_eq!(self.required_with_default.get_class(), GString::from("Node"));
+///         // Will always be valid and must be set via editor – an additional check is being run before ready to make sure that given value can't be null.
+///         let some_variant = self.editor_field.get_meta("SomeName");
+///     }
+/// }
+/// ```
```

Comment:
Please check the generated PR output per bot link -- the code needs horizontal scrolling, and some statements are not aligned :slightly_smiling_face: 

https://godot-rust.github.io/docs/gdext/pr-1051/godot/obj/struct.OnEditor.html